packages/amplify-category-analytics/src/analytics-resource-api.ts

```diff
@@ -9,34 +9,19 @@ import { addResource } from './provider-utils/awscloudformation/index';
 import { analyticsPush } from './commands/analytics';
 import { invokeAuthPush } from './plugin-client-api-auth';
 import { invokeNotificationsAPIGetAvailableChannelNames } from './plugin-client-api-notifications';
+import { pinpointHasInAppMessagingPolicy } from './utils/pinpoint-helper';
+import { getAnalyticsResources } from './utils/analytics-helper';
+import { analyticsMigrations } from './migrations';
 
 /**
  * Get all analytics resources. If resourceProviderService name is provided,
  * then only return resources matching the service.
  * @returns Array of resources in Analytics category (IAmplifyResource type)
  */
-export const analyticsPluginAPIGetResources = (resourceProviderServiceName?: string, context?: $TSContext): Array<IAnalyticsResource> => {
-  const resourceList: Array<IAnalyticsResource> = [];
-  const amplifyMeta = (context) ? context.exeInfo.amplifyMeta : stateManager.getMeta();
-  if (amplifyMeta?.[AmplifyCategories.ANALYTICS]) {
-    const categoryResources = amplifyMeta[AmplifyCategories.ANALYTICS];
-    Object.keys(categoryResources).forEach(resource => {
-      // if resourceProviderService is provided, then only return resources provided by that service
-      // else return all resources. e.g. Pinpoint, Kinesis
-      if (!resourceProviderServiceName || categoryResources[resource].service === resourceProviderServiceName) {
-        resourceList.push({
-          category: AmplifyCategories.ANALYTICS,
-          resourceName: resource,
-          service: categoryResources[resource].service,
-          region: categoryResources[resource]?.output?.Region,
-          id: categoryResources[resource]?.output?.Id,
-          output: categoryResources[resource]?.output,
-        });
-      }
-    });
-  }
-  return resourceList;
-};
+export const analyticsPluginAPIGetResources = (
```

Comment:
body moved as it was needed to be reused for migration logic
packages/amplify-category-analytics/src/utils/pinpoint-helper.ts